src/main/java/com/glencoesoftware/ldaptool/Search.java

```diff
@@ -0,0 +1,156 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.glencoesoftware.ldaptool;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ldap.core.DistinguishedName;
+import org.springframework.ldap.core.LdapTemplate;
+
+import ome.logic.LdapImpl;
+import ome.logic.LdapImpl.GroupLoader;
+import ome.model.meta.Experimenter;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.concurrent.Callable;
+
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParentCommand;
+
+/**
+ * @author Chris Allan <callan@glencoesoftware.com>
+ */
+@Command(
+    name = "search",
+    mixinStandardHelpOptions = true,
+    exitCodeOnExecutionException = 100
+)
+public class Search implements Callable<Integer>
+{
+    private static final Logger log =
+            LoggerFactory.getLogger(Search.class);
+
+    @ArgGroup(exclusive = true, multiplicity = "1")
+    SearchFor searchFor;
+
+    static class SearchFor {
+        @Option(names = "--all", description = "Print all users")
+        boolean all;
+
+        @Option(names = "--user", description = "Username to search")
+        String[] username;
+    }
+
+    @ParentCommand
+    private Main main;
+
+    Search() { }
+
+    @Override
+    public Integer call() throws Exception {
+        String referral = main.context.getProperty("omero.ldap.referral");
+        Field ignorePartialResultException =
+                LdapTemplate.class.getDeclaredField("ignorePartialResultException");
+        ignorePartialResultException.setAccessible(true);
+        log.info("Ignoring partial result exceptions? {}",
+                ignorePartialResultException.get(main.ldapTemplate));
+        log.info("Referral set to: '{}'", referral);
+
+        System.out.println("---");
+        if (searchFor.all) {
+            lookupAllUsers(main.ldapImpl, main.ldapTemplate);
+        } else {
+            for (String username : searchFor.username) {
+                try {
+                    Experimenter experimenter =
+                            main.ldapImpl.findExperimenter(username);
+                    lookupUser(main.ldapImpl, main.ldapTemplate, experimenter);
+                } catch (ome.conditions.ApiUsageException api) {
+                    System.err.println("no such user: " + searchFor.username);
```

Comment:
```suggestion
                    System.err.println("No such user: " + username);
```

This was not introduced by this refactoring but while testing non-existing LDAP users, I realized that this error message is uninformative given that `searchFor.username` is a string array.

Alternatively, we could also print the `ApiUsageException` which will read like `Cannot find unique user DistinguishedName...` in these scenarios